src/sentry/api/endpoints/organization_events.py

```diff
@@ -62,6 +66,27 @@ def get(self, request, organization):
 
         now = timezone.now()
 
+        # default to last 90 days
+        end = now
+        start = now - timedelta(days=90)
+
+        stats_period = request.GET.get('statsPeriod')
+        if stats_period is not None:
+            stats_period = parse_stats_period(request.GET['statsPeriod'])
+            if stats_period is None or stats_period < self.MIN_STATS_PERIOD or stats_period >= self.MAX_STATS_PERIOD:
+                return Response({'detail': 'Invalid statsPeriod'}, status=400)
+            start = now - stats_period
+        elif request.GET.get('start') or request.GET.get('end'):
+            if not all([request.GET.get('start'), request.GET.get('end')]):
+                return Response({'detail': 'start and end are both required'}, status=400)
+            try:
+                start = parse_datetime_string(request.GET['start'])
+                end = parse_datetime_string(request.GET['end'])
+            except InvalidQuery as exc:
+                return Response({'detail': exc.message}, status=400)
+            if start > end:
+                return Response({'detail': 'start must be before end'}, status=400)
```

Comment:
This logic looks like the kind of thing that could live in a rest framework serializer if you wanted to trim down the endpoint logic.

Reply:
we're probably going to replace this completely in the near future to support many more search terms, so i'd rather just leave how it is for now